doc: Trust Model Introduction

magowan opened the pull request:
Add a introduction to Trust Model.
Include references and links to prior art

This PR is one part of Issue confidential-containers/confidential-containers#117

Follow on Pull Requests will include sections corresponding to orthogonal ways of considering cloud native approaches relevent to the trust model

Signed-off-by: James Magowan <magowan@uk.ibm.com>

fitzthum reviewed (commented):
Generally unobjectionable. Tbh I think we could skip over a lot of stuff here. I'm not sure that we need to define what a trust model is, for instance. If someone reading the document really doesn't know, they can just google it. I guess it can't hurt to provide some intro, but I think the most important part of this doc so far is the Out of Scope section. That's where we introduce concrete information specific to this project.

magowan commented:
> Generally unobjectionable. Tbh I think we could skip over a lot of stuff here. I'm not sure that we need to define what a trust model is, for instance. If someone reading the document really doesn't know, they can just google it. I guess it can't hurt to provide some intro, but I think the most important part of this doc so far is the Out of Scope section. That's where we introduce concrete information specific to this project.

Yes I have scaled back the intro from early "draft", and agree  the important stuff is initially the out-of-scope for this but the main pieces come with the detail in follow on PR's . I am inclined to not reduce the content further , as I don't think it hurts and there may be different subtle takes on a threat model so can;t hurt to try to clarify what we draw on for ours.

magowan commented on TrustModel.md:
```diff
@@ -0,0 +1,113 @@
+   - Lifecycle
+   - Trusted Execution Boundary Location
+   - Secured v unsecured TEE Base Image
+-->
```
I only put this in so that people can see what the follow on PR's will contain, happy to just remove this from this PR if that is better?

dcmiddle replied:
I found it helpful to see this list since we split it out of #16.
In that earlier PR you referred to [guest] `boot image` in a few places. I take that to mean what you here call `base image`?
Also note that's only relevant to the pod/vm approach.

magowan replied:
Yes I'll leave the comment out list there then so everyone can be clear what is coming

jodh-intel replied:
If we plan to land this PR, I think the list should be moved to a GH issue (now) to allow more visible tracking of outstanding items, and easier and quicker updates to that list.

magowan replied:
The list does currently exist in this issue -> https://github.com/confidential-containers/documentation/issues/14
So happy to remove this and rely on issue

magowan replied:
I have create two new issues to give a fresh basis for the continued discussion
#28 covers a list of Trust model documentation to be completed
#29 covers a list of Threats Vectors/Profiles we should document (should use concepts and terminology agreed in model documentation)

magowan commented on TrustModel.md:
```diff
@@ -0,0 +1,113 @@
+For example: 
+
+- Confidential Computing Consortium (CCC) published 
+  "[A Technical Analysis of Confidential Computing](https://confidentialcomputing.io/wp-content/uploads/sites/85/2021/03/CCC-Tech-Analysis-Confidential-Computing-V1.pdf) [^1]" 
```
Yes bit of duplication, I need to choose one or the other!.
Open to whatever people feel, The footnotes probably made more sense when it was one long document and saved repeating a link but with the model heading towards separate documents for aspects of it, maybe it makes less sense to use footnotes.
What do people think , just links or footnotes?

bpradipt replied:
I don't think footnotes are needed when links are present.

jiazhang0 replied:
If a single link without other comment info is enough, just use it, even it will be repeated. If we need to refer to a specific page number or section name of material, footnote with page number or section name will be useful, or use this form:`The Security Assurance section, Page 31, of [CNCF Cloud Native Security Whitepaper](https://github.com/cncf/tag-security/blob/main/security-whitepaper/CNCF_cloud-native-security-whitepaper-Nov2020.pdf)`

magowan replied:
Removed the footnotes tidied up references

magowan commented on TrustModel.md:
Thanks, fixed

magowan commented on TrustModel.md:
Good point, added a bullet to consider this.

bpradipt commented on TrustModel.md:
nit TEE's => TEEs

magowan replied:
Fixed

bpradipt commented on TrustModel.md:
Nit:  Using the expanded form `Trusted Execution Environment (TEE)` for the first occurrence and consistently using TEE/TEEs for all other occurrences?

magowan replied:
Thanks, fixed

bpradipt commented:
@magowan  are you planning to include details about different threat actors for the CoCo scenario and also a (good-to-have) pictorial representation of the trust boundaries for the complete CoCo solution in separate PRs?

dcmiddle commented:
I think this is generally good as is.
With the other sections there's going to be sort of a lot to read to get to the punch line that
1. with pod isolation we draw a box around the guest
2. with container isolation we draw a box around the process.
3. actors in the control plane may be able to break into those boxes. (I haven't reviewed #24 yet)

What do you think about an informal section in the overview to give that intuition before getting into the formalities?

magowan commented:
> @magowan are you planning to include details about different threat actors for the CoCo scenario and also a (good-to-have) pictorial representation of the trust boundaries for the complete CoCo solution in separate PRs?

Yes, That is the goal , there is another PR up already regarding the Personas/Actors, but I will put other PRs together for other areas and then my intention is once we have a trust model, I can put up some threat Vectors set in the context of our trust model which can highlight how we solve them with CoCo.

magowan commented on TrustModel.md:
```diff
@@ -0,0 +1,117 @@
+execution (5.1 Goal)[^1].
+
+This means our trust and threat modelling should 
+- Focus on confidentiality and integrity of data and code.
```
Agree, taken your wording as better, thanks

bpradipt reviewed (approved):
/lgtm
Thanks @magowan

jiazhang0 commented on TrustModel.md:
Plz remove these empty lines.

magowan replied:
Done

jodh-intel commented on TrustModel.md:
```diff
@@ -0,0 +1,106 @@
+The [Trust Modeling for Security Architecture Development article](https://www.informit.com/articles/article.aspx?p=31546) 
+defines Trust Modeling as :
+
+    A trust model identifies the specific mechanisms that are necessary to respond to a specific 
+    threat profile.
+
+    A trust model must include implicit or explicit validation of an entity's identity or the 
+    characteristics necessary for a particular event or transaction to occur.
```
Nit: You could use a block quote for these quotes:

> A trust model identifies the specific mechanisms that are necessary to respond to a specific 
> threat profile.

> A trust model must include implicit or explicit validation of an entity's identity or the 
> characteristics necessary for a particular event or transaction to occur.

```suggestion
> A trust model identifies the specific mechanisms that are necessary to respond to a specific 
> threat profile.

> A trust model must include implicit or explicit validation of an entity's identity or the 
> characteristics necessary for a particular event or transaction to occur.
```

jodh-intel replied:
Also, where do we plan to list the approach(es) taken for entity validation? Can this be linked to or a "TODO: See GitHub issue $url" added?

magowan replied:
Added the blockquotes and added Explicit validation of an entity's identity (attestation) to the Issue #28

jodh-intel commented on TrustModel.md:
```diff
@@ -0,0 +1,106 @@
+[trust boundary](https://en.wikipedia.org/wiki/Trust_boundary) describes a location where 
+ program data or execution changes its level of "trust", or where two principals with different 
+ capabilities exchange data or commands. Specific to Confidential Containers is the trust 
+ boundary that corresponds to the boundary of the Trusted Execution Environment (TEE). Generally, 
```
Can we drop "Generally"?

magowan replied:
Done

jodh-intel commented on TrustModel.md:
```diff
@@ -0,0 +1,106 @@
+components and architecture deliver the security principles expected for cloud native 
+confidential computing. It provides the solid foundations and unifying security principles 
+against which we can assess architecture and implementation ideas and discussions.
+
```
How about adding a heading here?

```suggestion
## Trust Model Definition
```

magowan replied:
Done

jodh-intel commented on TrustModel.md:
```diff
@@ -0,0 +1,106 @@
+
+    A trust model must include implicit or explicit validation of an entity's identity or the 
+    characteristics necessary for a particular event or transaction to occur.
+
```
Again, I wonder if it's worth adding a "why is this important?" type heading here.

magowan replied:
Added heading

jodh-intel commented on TrustModel.md:
```diff
@@ -0,0 +1,106 @@
+ boundary that corresponds to the boundary of the Trusted Execution Environment (TEE). Generally, 
+ the TEE side of the trust boundary will be hardened to prevent the violation of the trust 
+ boundary.
+
```
Is this paragraph the "plan" / "mission statement" / TODO? Again, this would be clearer with a heading I think.

magowan replied:
Added Heading

jodh-intel commented on TrustModel.md:
```diff
@@ -0,0 +1,106 @@
+defines Trust Modeling as :
+
+    A trust model identifies the specific mechanisms that are necessary to respond to a specific 
+    threat profile.
```
Where do we plan to list all profiles targetted to this project? Can this be linked to or a "TODO: See GitHub issue $url" added?

magowan replied:
Fresh Issues
#28 Lists documentation required for Trust Model
#29 Lists currents thoughts on Threat Vectors/Profiles to be described (using Trust Model terminology)

jodh-intel commented on TrustModel.md:
```diff
@@ -0,0 +1,106 @@
+   - Secured v unsecured TEE Base Image
+-->
+
+Further documentation will highlight specific threat vectors in detail, considering risk, 
```
Can we list the basic threat vectors at this stage, or reference another file (even a "stub" markdown file that could be created with this PR)?

magowan replied:
I added a stub file which has a link to the Issue #29 which holds a list to be documented

jodh-intel reviewed (commented):
Thanks @magowan. A few comments...

magowan committed:
docs: Trust Model Introduction

Add a introduction to Trust Model (Part of Issue #28).
Include references and links to prior art

Follow on Pull Requests will include sections corresponding to orthogonal ways of considering cloud native approaches relevent to the trust model
Specifically including
   - Personas
   - Lifecycle
   - Trusted Execution Boundary Location
   - Secured v unsecured TEE Base Image

Signed-off-by: James Magowan <magowan@uk.ibm.com>

magowan commented:
@jodh-intel  Thanks for review.
I think I have covered your review feedback.
If you are happy with my response/changes then I think we are ready to merge.
(But happy to take any other comments on board)

jodh-intel reviewed (approved):
Thanks @magowan.

lgtm
